### PR TITLE
Nuke the vast majority of clang warnings

### PR DESCRIPTION
**Pull request by andyp1per**
This removes most clang warnings on macos. Still warnings about std::move() but I am not sure how to fix those

### CODE REVIEW

File: libraries/AP_Frsky_Telem/AP_Frsky_SPort.cpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ extern const AP_HAL::HAL& hal;
 bool AP_Frsky_SPortParser::should_process_packet(const uint8_t *packet, bool discard_duplicates)
 {
     // check for duplicate packets
-    if (discard_duplicates && _parse_state.last_packet != nullptr) {
```
Comment (andyp1per) [inline libraries/AP_Frsky_Telem/AP_Frsky_SPort.cpp:249]: @yaapu confirmed that this check is redundant

File: libraries/AP_Common/AP_Common.h [hunk 2]
```diff
@@ -33,6 +33,13 @@
 // used to mark a function that may be unused in some builds
 #define UNUSED_FUNCTION __attribute__((unused))
 
+// used to mark an attribute that may be unused in some builds
+#ifdef __clang__
+#define UNUSED_ATTRIBUTE __attribute__((unused))
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_Common/AP_Common.h:38]: Fixed

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by timtuxworth** [comment]
This is awesome, but still some left (built for plane/sitl)

```
[333/864] Compiling libraries/AP_Notify/DiscoLED.cpp
../../libraries/AP_Notify/Display.cpp:337:50: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            _driver = Display_SSD1306_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                 ^
../../libraries/AP_Notify/Display.cpp:337:50: note: remove std::move call here
            _driver = Display_SSD1306_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                 ^~~~~~~~~~                                                   ~
../../libraries/AP_Notify/Display.cpp:341:49: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
            _driver = Display_SH1106_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                ^
../../libraries/AP_Notify/Display.cpp:341:49: note: remove std::move call here
            _driver = Display_SH1106_I2C::probe(std::move(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR)));
                                                ^~~~~~~~~~                                                   ~
2 warnings generated.

[334/864] Compiling libraries/AP_Notify/AP_BoardLED.cpp
[335/864] Compiling libraries/AP_Notify/DiscreteRGBLed.cpp
[336/864] Compiling libraries/AP_Notify/NCP5623.cpp
[337/864] Compiling libraries/AP_Notify/Display_SH1106_I2C.cpp
[338/864] Compiling libraries/AP_Notify/NeoPixel.cpp
[339/864] Compiling libraries/AP_Notify/ProfiLED.cpp
[340/864] Compiling libraries/AP_Notify/ToshibaLED_I2C.cpp
../../libraries/AP_Notify/OreoLED_I2C.cpp:59:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.i2c_mgr->get_device(_bus, OREOLED_BASE_I2C_ADDR));
           ^
../../libraries/AP_Notify/OreoLED_I2C.cpp:59:12: note: remove std::move call here
    _dev = std::move(hal.i2c_mgr->get_device(_bus, OREOLED_BASE_I2C_ADDR));
           ^~~~~~~~~~                                                    ~
1 warning generated.

[341/864] Compiling libraries/AP_Notify/PixRacerLED.cpp
[342/864] Compiling libraries/AP_Notify/Led_Sysfs.cpp
[343/864] Compiling libraries/AP_Notify/ScriptingLED.cpp
[344/864] Compiling libraries/AP_Notify/SerialLED.cpp
[345/864] Compiling libraries/AP_Notify/MMLPlayer.cpp
../../libraries/AP_Notify/NCP5623.cpp:71:16: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
        _dev = std::move(hal.i2c_mgr->get_device(_bus, addrs[i]));
               ^
../../libraries/AP_Notify/NCP5623.cpp:71:16: note: remove std::move call here
        _dev = std::move(hal.i2c_mgr->get_device(_bus, addrs[i]));
               ^~~~~~~~~~                                       ~
1 warning generated.

[346/864] Compiling libraries/AP_Notify/ToneAlarm.cpp
[347/864] Compiling libraries/AP_Notify/SITL_SFML_LED.cpp
[348/864] Compiling libraries/AP_Notify/ExternalLED.cpp
[349/864] Compiling libraries/AP_Notify/RCOutputRGBLed.cpp
../../libraries/AP_Notify/ToshibaLED_I2C.cpp:50:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.i2c_mgr->get_device(_bus, TOSHIBA_LED_I2C_ADDR));
           ^
../../libraries/AP_Notify/ToshibaLED_I2C.cpp:50:12: note: remove std::move call here
    _dev = std::move(hal.i2c_mgr->get_device(_bus, TOSHIBA_LED_I2C_ADDR));
           ^~~~~~~~~~                                                   ~
1 warning generated.

[350/864] Compiling libraries/AP_Notify/RGBLed.cpp
../../libraries/AP_Notify/ProfiLED.cpp:77:12: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    _dev = std::move(hal.spi->get_device("profiled"));
           ^
../../libraries/AP_Notify/ProfiLED.cpp:77:12: note: remove std::move call here
    _dev = std::move(hal.spi->get_device("profiled"));
           ^~~~~~~~~~                               ~
In file included from ../../libraries/AP_Notify/ProfiLED.cpp:17:
../../libraries/AP_Notify/ProfiLED.h:53:17: warning: private field '_last_init_ms' is not used [-Wunused-private-field]
    uint32_t    _last_init_ms;
                ^
2 warnings generated.

```

```
[406/864] Compiling libraries/AP_RPM/AP_RPM.cpp
../../libraries/AP_RCProtocol/spm_srxl.cpp:78:36: warning: suggest braces around initialization of subobject [-Wmissing-braces]
SrxlTelemetryData srxlTelemData = {0};
                                   ^
                                   {}
../../libraries/AP_RCProtocol/spm_srxl.cpp:83:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
static SrxlDevice srxlThisDev = {0};
                                 ^
                                 {}
../../libraries/AP_RCProtocol/spm_srxl.cpp:91:36: warning: suggest braces around initialization of subobject [-Wmissing-braces]
static SrxlReceiverStats srxlRx = {0};
                                   ^
                                   {}
../../libraries/AP_RCProtocol/spm_srxl.cpp:91:36: warning: suggest braces around initialization of subobject [-Wmissing-braces]
static SrxlReceiverStats srxlRx = {0};
                                   ^
                                   {}
../../libraries/AP_RCProtocol/spm_srxl.cpp:1307:26: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    SrxlFullID retVal = {0};
                         ^
                         {}
5 warnings generated.


```

```
[730/864] Compiling libraries/AP_Math/quaternion.cpp
../../libraries/AP_Baro/AP_Baro.cpp:665:43: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                          std::move(GET_I2C_DEVICE(_ext_bus, HAL_BARO_MS5611_I2C_ADDR))));
                                          ^
../../libraries/AP_Baro/AP_Baro.cpp:726:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_BMP085_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:732:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_BMP280_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:734:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_BMP280_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:740:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_SPL06_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:742:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_SPL06_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:748:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_BMP388_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:750:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_BMP388_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:756:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_MS5611_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:758:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_MS5611_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:764:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_MS5607_I2C_ADDR)),
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:771:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_MS5637_I2C_ADDR)),
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:778:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_FBM320_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:780:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_FBM320_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:786:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_DPS280_I2C_ADDR))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:788:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_DPS280_I2C_ADDR2))));
                                              ^
../../libraries/AP_Baro/AP_Baro.cpp:794:47: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
                                              std::move(GET_I2C_DEVICE(i, HAL_BARO_LPS25H_I2C_ADDR))));
                                              ^
17 warnings generated.

```

```
[863/864] Compiling ArduPlane/mode_takeoff.cpp
../../ArduPlane/commands_logic.cpp:1023:23: warning: absolute value function 'fabsf' given an argument of type 'double' but has parameter of type 'float' which may cause truncation of value [-Wabsolute-value]
                if (((fabsf(cmd.content.location.get_distance(current_loc) - radius) > 5.0f) &&
                      ^
../../ArduPlane/commands_logic.cpp:1023:23: note: use function 'std::abs' instead
                if (((fabsf(cmd.content.location.get_distance(current_loc) - radius) > 5.0f) &&
                      ^~~~~
                      std::abs
1 warning generated.

../../ArduPlane/mode_qrtl.cpp:15:30: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                             ^
../../ArduPlane/mode_qrtl.cpp:15:30: note: use function 'std::abs' instead
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                             ^~~~~
                             std::abs
../../ArduPlane/mode_qrtl.cpp:15:64: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                                                               ^
../../ArduPlane/mode_qrtl.cpp:15:64: note: use function 'std::abs' instead
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                                                               ^~~~~
                                                               std::abs
../../ArduPlane/mode_qrtl.cpp:76:30: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                             ^
../../ArduPlane/mode_qrtl.cpp:76:30: note: use function 'std::abs' instead
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                             ^~~~~
                             std::abs
../../ArduPlane/mode_qrtl.cpp:76:64: warning: using floating point absolute value function 'fabsf' when argument is of integer type [-Wabsolute-value]
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                                                               ^
../../ArduPlane/mode_qrtl.cpp:76:64: note: use function 'std::abs' instead
    const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
                                                               ^~~~~
                                                               std::abs
4 warnings generated.

```

**Pull request merged**